### PR TITLE
fix: resolve DataCloudArray data loss after ResultSet exhaustion

### PR DESCRIPTION
**Pull request by KaviarasuSakthivadivel**
- Refactor DataCloudArray to extract and store data during construction
- Fix JDBC array indexing to use proper 1-based indexing
- Add comprehensive tests for string arrays with NULL values

Fixes data loss issue where ValueVector data was lost after ResultSet lifecycle ended, following PostgreSQL JDBC ArrayImpl pattern.

### CODE REVIEW

File: jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java [hunk 1]
```diff
@@ -51,7 +82,7 @@ public Object getArray(Map<String, Class<?>> map) throws SQLException {
         if (map != null) {
             throw new DataCloudJDBCException(NOT_SUPPORTED_IN_DATACLOUD_QUERY, SqlErrorCodes.FEATURE_NOT_SUPPORTED);
         }
-        return getArrayNoBoundCheck(this.dataVector, this.startOffset, this.valuesCount);
+        return this.data.clone();
```
Comment (mkaufmann) [inline jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java:85]: This should not be required, if a user modifies the array it's their own faul. Having an additional clone would pessimize performance

File: jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java [hunk 2]
```diff
@@ -94,17 +127,15 @@ public void free() {
...
-        if (index < 0 || index + count > this.startOffset + this.valuesCount) {
-            throw new ArrayIndexOutOfBoundsException();
+        // Check against actual stored data size using 1-based JDBC indexing
+        if (index < 1 || index > this.data.length) {
+            throw new ArrayIndexOutOfBoundsException("Index " + index + " is out of bounds for array size "
+                    + this.data.length + " (JDBC arrays use 1-based indexing)");
+        }
+        if (index + count - 1 > this.data.length) {
+            throw new ArrayIndexOutOfBoundsException("Index " + index + " + count " + count + " exceeds array size "
+                    + this.data.length + " (JDBC arrays use 1-based indexing)");
```
Comment (mkaufmann) [inline jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java:138]: I think the following formulation should be simpler as it follows the normal 0 based bound check logic
```suggestion
        long dataIndex = index - 1;
        // Check against actual stored data size using 1-based JDBC indexing
        if (dataIndex < 0 || dataIndex >= this.data.length) {
            throw new ArrayIndexOutOfBoundsException("Index " + index + " is out of bounds for array size "
                    + this.data.length + " (JDBC arrays use 1-based indexing)");
        }
        if ((dataIndex + count) >= this.data.length) {
            throw new ArrayIndexOutOfBoundsException("Index " + index + " + count " + count + " exceeds array size "
                    + this.data.length + " (JDBC arrays use 1-based indexing)");
```

File: jdbc-core/src/test/resources/simplelogger.properties [hunk 3]
```diff
@@ -1,2 +1,2 @@
 org.slf4j.simpleLogger.logFile=System.out
-org.slf4j.simpleLogger.defaultLogLevel=warn
+org.slf4j.simpleLogger.defaultLogLevel=info
```
Comment (mkaufmann) [inline jdbc-core/src/test/resources/simplelogger.properties:2]: Nit: Remove

File: jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArrayTest.java [hunk 4]
```diff
@@ -108,6 +113,121 @@ void testShouldThrowIfGetArrayHasIncorrectIndex() {
...
+
+    @SneakyThrows
+    @Test
+    void testGetArrayWithZeroCount() {
+        // Test case with valuesCount = 0
+        val values = ImmutableList.of(1, 2, 3);
+        dataVector = extension.createIntVector(values);
+        val dataCloudArray = new DataCloudArray(dataVector, 0, 0);
+        val array = (Object[]) dataCloudArray.getArray();
+        collector.assertThat(array).isEmpty();
```
Comment (mkaufmann) [inline jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArrayTest.java:136]: SHould dataCloudArray.getArray(0,0) work in this scenario?

File: jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java (hunk 1 above)
Comment (KaviarasuSakthivadivel) [reply jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java:85]: Yes, I think we can just return the data here. I will modify it. Thank you for catching this, I was having this as a test.

File: jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java (hunk 2 above)
Comment (KaviarasuSakthivadivel) [reply jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/accessor/impl/DataCloudArray.java:139]: Modified the logic, this is simpler. Thank you, Moritz.

**Review by mkaufmann — APPROVED** [review]
